application/libraries/PluginManager/PluginManager.php

```diff
@@ -322,6 +322,9 @@ public function scanPlugins($includeInstalledPlugins = false)
                                     ];
                                     $saveResult = Plugin::setPluginLoadError($plugin, $pluginName, $error);
                                     if (!$saveResult) {
+                                        if(App()->getConfig('debug') >= 2) {
+                                            throw $ex;
+                                        }
                                         // This only happens if database save fails.
```

Comment:
I would add a comment here saying that this will happen only if debug is not set to 2, right?

Reply:
2 comments

```
                                        // If handlePluginLoadError return 0 because debug is set
                                        if (App()->getConfig('debug') >= 2) {
                                            throw $ex;
                                        }
                                        // If handlePluginLoadError fail without debug : have a DB related issue
```